tests/checkpointing/test_model_checkpoint.py

```diff
@@ -261,6 +261,40 @@ def test_model_checkpoint_format_checkpoint_name(tmpdir):
     assert ckpt_name == filepath / 'test-epoch=3-step=2.ckpt'
 
 
+def test_model_checkpoint_file_extension(tmpdir):
+    
+    # tests that format_checkpoint_name uses the user-defined FILE_EXTENSION
+    ckpt_name = ModelCheckpoint(monitor='early_stop_on', dirpath='.').format_checkpoint_name(0, 1, {}) 
+    ModelCheckpoint.FILE_EXTENSION = '.tpkc'
```

Comment:
pls do it by inheriting `ModelCheckpoint`.
Also checking with `format_checkpoint_name` is not required here. You can just test the final output filenames with `max_epochs=2, save_top_k=1, save_last=True`. That would be enough I think.

Reply:
agree, inherit then you don't need to reset at the end and keeps the class untouched for the rest of the tests. 
also a simple unit test on the checkpoint class is sufficient or if you really want the integration test then I suggest running only one batch instead of a full epoch.